scr/main/java/bittercode/constant/db/UsersDBConstants.java

```diff
@@ -0,0 +1,15 @@
+package com.bittercode.constant.db;
+
+public interface UsersDBConstants {
+
+	public static String TABLE_USERS = "users";
+	
+	public static String COLUMN_USERNAME = "username";
+	public static String COLUMN_PASSWORD = "password";
+	public static String COLUMN_FIRSTNAME = "firstname";
+	public static String COLUMN_LASTNAME = "lastname";
+	public static String COLUMN_ADDRESS = "address";
+	public static String COLUMN_PHONE = "phone";
+	public static String COLUMN_MAILID = "mailid";
+	public static String COLUMN_USERTYPE = "usertype";

```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider making constant fields final</b></div>
<div id="fix">

Consider making the constant fields `final` to ensure they cannot be modified after initialization. Constants should be immutable. A similar issue was also found in scr/main/java/bittercode/constant/db/BooksDBConstants.java (line 5,7-11).
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -5,10 +5,10 @@
 -	public static String TABLE_USERS = "users";
 +	public static final String TABLE_USERS = "users";
 -	public static String COLUMN_USERNAME = "username";
 +	public static final String COLUMN_USERNAME = "username";
 ...
```
</div>
</details>
</div>

<div id="suggestion">
<div id="issue"><b>Consider renaming password-related database column constant</b></div>
<div id="fix">

It appears this constant directly references a password column in the database schema. While this might be needed for database operations, consider using a more generic name to avoid exposing database structure details. This follows security best practices of reducing information disclosure about sensitive fields.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -7,7 +7,7 @@ public interface UsersDBConstants {
  	public static String COLUMN_USERNAME = "username";
 -	public static String COLUMN_PASSWORD = "password";
 +	public static String COLUMN_CREDENTIAL = "password";
  	public static String COLUMN_FIRSTNAME = "firstname";
  	public static String COLUMN_LASTNAME = "lastname";
  	public static String COLUMN_ADDRESS = "address";
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/1#issuecomment-2514333017>#855223</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged